sql-plugin/src/main/scala/com/nvidia/spark/rapids/SparkShims.scala

```diff
@@ -252,6 +252,11 @@ trait SparkShims {
   def isCustomReaderExec(x: SparkPlan): Boolean
 
   def aqeShuffleReaderExec: ExecRule[_ <: SparkPlan]
+
+  /** Return list of operators in the plan that match the predicate */
+  def findOperators(plan: SparkPlan, predicate: SparkPlan => Boolean): Seq[SparkPlan]
+
+  def shouldAssertIsOnTheGpu(plan: SparkPlan): Boolean
```

Comment:
Can we get a little more explanation about this? It is both obvious what it is doing, and very non-obvious because it is combined with other common patterns.  We could also rename it so it includes extra, or something like that.

Reply:
I pushed some javadocs. I am happy to change the name too but could not find anything I was happy with so far.